profitbricks/src/main/java/org/jclouds/profitbricks/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,395 @@
+                    .imagePassword(password)
+                    .name(format("%s-disk-%d", name, i++))
+                    .size(volume.getSize())
+                    .build());
+
+            waitDcUntilAvailable.apply(dataCenterId);
+            storageIds.add(id);
+            logger.trace(">> provisioning complete for storage. returned id='%s'", id);
+         } catch (Exception ex) {
+            if (i - 1 == 1) // if first storage (one with image) provisioning fails; stop method
+               throw Throwables.propagate(ex);
+            logger.warn(ex, ">> failed to provision storage. skipping..");
+         }
+
+      Double cores = 0d;
+      for (Processor processor : hardware.getProcessors())
+         cores += processor.getCores();
+
+      // provision server and connect boot storage (first provisioned)
+      String serverId = null;
+      try {
+         String storageBootDeviceId = Iterables.get(storageIds, 0); // must have atleast 1
+         Server.Request.CreatePayload serverRequest = Server.Request.creatingBuilder()
+                 .dataCenterId(dataCenterId)
+                 .name(name)
+                 .bootFromStorageId(storageBootDeviceId)
+                 .cores(cores.intValue())
+                 .ram(hardware.getRam())
+                 .availabilityZone(AvailabilityZone.AUTO)
+                 .hasInternetAccess(true)
+                 .build();
+         logger.trace("<< provisioning server '%s'", serverRequest);
+         serverId = api.serverApi().createServer(serverRequest);
+         waitDcUntilAvailable.apply(dataCenterId);
+
+         logger.trace(">> provisioning complete for server. returned id='%s'", serverId);
+      } catch (Exception ex) {
+         logger.error(ex, ">> failed to provision server. rollbacking..");
+         destroyStorages(storageIds, dataCenterId);
+         throw Throwables.propagate(ex);
+      }
+
+      // connect the rest of storages to server; delete if fails
+      final int storageCount = storageIds.size();
+      for (int j = 1; j < storageCount; j++) { // skip first; already connected
+         String storageId = storageIds.get(j);
+         try {
+            logger.trace("<< connecting storage '%s' to server '%s'", storageId, serverId);
+            api.storageApi().connectStorageToServer(
+                    Storage.Request.connectingBuilder()
+                    .storageId(storageId)
+                    .serverId(serverId)
+                    .build());
+
+            waitDcUntilAvailable.apply(dataCenterId);
+            logger.trace(">> storage connected.");
+         } catch (Exception ex) {
+            // delete unconnected storage
+            logger.warn(ex, ">> failed to connect storage '%s'. deleting..", storageId);
+            waitDcUntilAvailable.apply(dataCenterId);
+            destroyStorage(storageId, dataCenterId);
+         }
+      }
+
+      waitDcUntilAvailable.apply(dataCenterId);
```

Comment:
Hmm. Yeah it's no longer needed. I was too paranoid of the datacenter not being ready I guess. Haha